src/app/qgsmaptooladdcircle.cpp

```diff
@@ -50,6 +50,13 @@ void QgsMapToolAddCircle::deactivate()
       lineString->addZValue( point.z() );
       break;
     }
+    if ( QgsWkbTypes::hasM( point.wkbType() ) &&
+         point.z() != defaultMValue() )
```

Comment:
shouldn't that be ?
```suggestion
         point.m() != defaultMValue() )
```
not caught by tests ?

Reply:
Right!
Yes it is not tested for now (as for the Z). It will be tested soon after my others PR on M/Z editing.

Reply:
Mmm, I should split up my tests and introduce the snapping tests for circle into this PR